From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: JiaY-shi <shi05275@163.com>
Date: Mon, 31 Aug 2026 02:20:00 +0800
Subject: [PATCH] arm: mediatek: add MT7981 distro boot environment

Provide standard load addresses and distro boot targets for MT7981 boards.

Signed-off-by: JiaY-shi <shi05275@163.com>
---
 include/configs/mt7981.h | 33 +++++++++++++++++++++++++++++++++
 1 file changed, 33 insertions(+)

--- a/include/configs/mt7981.h
+++ b/include/configs/mt7981.h
@@ -8,4 +8,37 @@
 #ifndef __MT7981_H
 #define __MT7981_H
 
+#include <linux/sizes.h>
+
+#if IS_ENABLED(CONFIG_CMD_USB)
+#define BOOT_TARGET_USB(func)	func(USB, usb, 0)
+#else
+#define BOOT_TARGET_USB(func)
+#endif
+
+#if IS_ENABLED(CONFIG_CMD_MMC)
+#define BOOT_TARGET_MMC(func)	func(MMC, mmc, 0)
+#else
+#define BOOT_TARGET_MMC(func)
+#endif
+
+#define BOOT_TARGET_DEVICES(func)	\
+	BOOT_TARGET_USB(func)		\
+	BOOT_TARGET_MMC(func)
+
+#define ENV_MEM_LAYOUT_SETTINGS		\
+	"scriptaddr=0x50000000\0"	\
+	"script_size_f=0x2000\0"	\
+	"kernel_addr_r=0x52000000\0"	\
+	"fdt_addr_r=0x62000000\0"	\
+	"ramdisk_addr_r=0x62180000\0"
+
+#include <config_distro_bootcmd.h>
+
+#define CFG_EXTRA_ENV_SETTINGS		\
+	"fdtfile=" CONFIG_DEFAULT_FDT_FILE "\0"	\
+	ENV_MEM_LAYOUT_SETTINGS		\
+	BOOTENV				\
+	"bootcmd=run distro_bootcmd\0"
+
 #endif
